checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,351 @@
+
+          CFStore elseStore = result.getElseStore();
+          lubWithStoreValue(elseStore, targetExpr, defaultType);
+        } else {
+          CFStore store = result.getRegularStore();
+          lubWithStoreValue(store, targetExpr, defaultType);
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Computes the LUB of the current value in the store for expr, if it exists, and defaultType.
+   * Inserts that LUB into the store as the new value for expr.
+   *
+   * @param store a CFStore
+   * @param expr an expression that might be in the store
+   * @param defaultType the default type of the expression's static type
+   */
+  private void lubWithStoreValue(CFStore store, JavaExpression expr, AnnotationMirror defaultType) {
+    CFValue value = store.getValue(expr);
+    CFValue defaultTypeAsValue = analysis.createSingleAnnotationValue(defaultType, expr.getType());
+    CFValue newValue;
+    if (value == null) {
+      newValue = defaultTypeAsValue;
+    } else {
+      newValue = value.leastUpperBound(defaultTypeAsValue);
+    }
+    store.clearValue(expr);
+    store.insertValue(expr, newValue);
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitTernaryExpression(
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * Adds newAnno as the value for target to all stores contained in result.
+   *
+   * @param result a TransferResult containing one or more stores
+   * @param target a JavaExpression whose type is being modified
+   * @param newAnno the new type for target
+   */
+  public void insertIntoStores(
+      TransferResult<CFValue, CFStore> result, JavaExpression target, AnnotationMirror newAnno) {
+    if (result.containsTwoStores()) {
+      CFStore thenStore = result.getThenStore();
+      CFStore elseStore = result.getElseStore();
+      thenStore.insertValue(target, newAnno);
+      elseStore.insertValue(target, newAnno);
+    } else {
+      CFStore store = result.getRegularStore();
+      store.insertValue(target, newAnno);
+    }
+  }
+
+  /**
+   * Returns the arguments of the @CreatesObligation annotation on the invoked method, as
+   * JavaExpressions. Returns the empty set if the given method has no @CreatesObligation
+   * annotation.
+   *
+   * <p>If any expression is unparseable, this method reports an error and returns the empty set.
+   *
+   * @param n a method invocation
+   * @param atypeFactory the type factory to report errors and parse the expression string
+   *     recomputation
```

Comment:
What does "recomputation" mean here?

Reply:
I genuinely have no idea. I think it must have been a typo.